packages/backend/src/graphql/queries/test-connection.ts

```diff
@@ -3,25 +3,56 @@ import App from '@/models/app'
 import Context from '@/types/express/context'
 
 type Params = {
-  id: string
-  data: object
+  connectionId: string
+  // when this is supplied, we check that the verify that the webhook url is properly set as well
+  stepId?: string
+}
+
+type TestConnectionOutput = {
+  connectionVerified: boolean
+  webhookVerified?: boolean
+  message?: string
 }
 
 const testConnection = async (
   _parent: unknown,
   params: Params,
   context: Context,
-) => {
+): Promise<TestConnectionOutput> => {
   let connection = await context.currentUser
     .$relatedQuery('connections')
     .findOne({
-      id: params.id,
+      id: params.connectionId,
     })
     .throwIfNotFound()
 
   const app = await App.findOneByKey(connection.key, false)
-  const $ = await globalVariable({ connection, app })
+  let $ = await globalVariable({ connection, app })
+
+  let step
+  if (params.stepId) {
+    step = await context.currentUser
```

Comment:
nit: In theory we only need this extra query for triggers with `verifyHook`. For micro-optimization we can move this down to below the `if (!isStillVerified || !params.stepId)` block.

Reply:
yeap this is true. But on the frontend, params.stepId will _not_ be sent if the app/step doesnt support webhook registration.